src/os/bluestore/BlueStore.cc

```diff
@@ -5023,32 +5034,38 @@ int BlueStore::_set_cache_sizes()
 
 int BlueStore::write_meta(const std::string& key, const std::string& value)
 {
-  bluestore_bdev_label_t label;
   string p = path + "/block";
-  int r = _read_bdev_label(cct, p, &label);
-  if (r < 0) {
-    return ObjectStore::write_meta(key, value);
+  if (bdev_label_valid_locations.empty()) {
+    _read_main_bdev_label(cct, p, &bdev_label,
+      &bdev_label_valid_locations, &bdev_label_multi, &bdev_label_epoch);
+  }
+  if (!bdev_label_valid_locations.empty()) {
+    bdev_label.meta[key] = value;
+    if (bdev_label_multi) {
+      bdev_label_epoch++;
+      bdev_label.meta["epoch"] = std::to_string(bdev_label_epoch);
+    }
+    int r = _write_bdev_label(cct, p, bdev_label, bdev_label_valid_locations);
+    ceph_assert(r == 0);
   }
-  label.meta[key] = value;
-  r = _write_bdev_label(cct, p, label);
-  ceph_assert(r == 0);
   return ObjectStore::write_meta(key, value);
 }
```

Comment:
What if `bdev_label_valid_locations.empty() == true`?

Reply:
We need to skip writing to bdev label if no bdev label was available, because we write "type=bluestore" in mkfs before creation of bdev label.
Granted, its weird.